token-swap/program/src/constraints.rs

```diff
@@ -44,7 +44,15 @@ impl<'a> SwapConstraints<'a> {
 
     /// Checks that the provided curve is valid for the given constraints
     pub fn validate_fees(&self, fees: &Fees) -> Result<(), ProgramError> {
-        if self.fees == fees {
+        // The owner and host fees are fixed. The trade fees can be variable
+        // depending on the swap.
+        if self.fees.owner_trade_fee_numerator == fees.owner_trade_fee_numerator
+            && self.fees.owner_trade_fee_denominator == fees.owner_trade_fee_denominator
+            && self.fees.owner_withdraw_fee_numerator == fees.owner_withdraw_fee_numerator
+            && self.fees.owner_withdraw_fee_denominator == fees.owner_withdraw_fee_denominator
+            && self.fees.host_fee_numerator == fees.host_fee_numerator
+            && self.fees.host_fee_denominator == fees.host_fee_denominator
```

Comment:
@gyeiv I have checked the Fee struct in the `fees.rs`(https://github.com/EIVentures/solana-program-library/blob/12551911d250e37c9a2b5df4681f8f8b195b7434/token-swap/program/src/curve/fees.rs#L13).
I think we need to check the `trade_fee_numerator` and `trade_fee_denominator`.

Reply:
This is exactly what we don't want. The reason is that `validate_fees` is only called on swap initialization. If we were to also check the trade fees, that would prevent us from setting them as we want for different markets.